packages/drivers/odsp-driver/src/test/odspDriverUrlResolver2Tests.spec.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable max-len */
+/* eslint-disable dot-notation */
+
+import { strict as assert } from "assert";
+import sinon from "sinon";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { OdspDriverUrlResolver2 } from "../odspDriverUrlResolver2";
+import { SharingLinkScopeFor, SharingLinkTokenFetcher } from "../tokenFetch";
+import { getHashedDocumentId } from "../odspUtils";
+import { createOdspUrl } from "../createOdspUrl";
+import * as graph from "../graph";
+import { getLocatorFromOdspUrl, storeLocatorInOdspUrl } from "../odspFluidFileLink";
+import { IOdspResolvedUrl, SharingLinkHeader } from "../contracts";
+
+describe("OdspUrlResolver3", () => {
+    const siteUrl = "https://microsoft.sharepoint-df.com/siteUrl";
+    const driveId = "driveId";
+    const itemId = "fileId";
+    const dataStorePath = "dataStorePath";
+    const fileName = "fileName";
+    const sharelink = "https://microsoft.sharepoint-df.com/site/SHARELINK";
+    let urlResolver: OdspDriverUrlResolver2;
+
+    beforeEach(() => {
+        const shareLinkTokenFetcher: SharingLinkTokenFetcher =
+            async (siteURL: string, scopeFor: SharingLinkScopeFor, refresh: boolean) => "SharingLinkToken";
+            urlResolver = new OdspDriverUrlResolver2(shareLinkTokenFetcher);
+    });
+
+    async function mockFetch<T>(response: Promise<string>, callback: () => Promise<T>): Promise<T> {
+        const fetchStub = sinon.stub(graph, "getShareLink");
+        fetchStub.returns(response);
+        try {
+            return await callback();
+        } finally {
+            fetchStub.restore();
+        }
+    }
+
+    it("resolve - Should resolve nav link correctly", async () => {
+        // Base64 encoded and then URI encoded string: d=driveId&f=fileId&c=dataStorePath&s=siteUrl&fluid=1
+        const url: string =
+        "https://microsoft.sharepoint-df.com/test?&nav=cz0lMkZzaXRlVXJsJmQ9ZHJpdmVJZCZmPWZpbGVJZCZjPWRhdGFTdG9yZVBhdGgmZmx1aWQ9MQ%3D%3D";
+        const resolvedUrl = await urlResolver.resolve({ url });
+        assert.strictEqual(resolvedUrl.driveId, driveId, "Drive id should be equal");
+        assert.strictEqual(resolvedUrl.siteUrl, siteUrl, "SiteUrl should be equal");
+        assert.strictEqual(resolvedUrl.itemId, itemId, "Item id should be absent");
+        assert.strictEqual(resolvedUrl.hashedDocumentId, getHashedDocumentId(driveId, itemId), "Doc id should be equal");
+        assert(resolvedUrl.endpoints.snapshotStorageUrl !== undefined, "Snapshot url should not be empty");
+    });
+
+    it("resolve - Should resolve odsp driver url correctly", async () => {
+        const url: string = createOdspUrl(siteUrl, driveId, itemId, dataStorePath);
```

Comment:
can we compare that url to the one in previous test? That way we are testing actual representation and conversions in both direcitons

Reply:
Added one more test for this.